torchgeo/datasets/utils.py

```diff
@@ -598,6 +600,25 @@ def unbind_samples(sample: Dict[Any, Sequence[Any]]) -> List[Dict[Any, Any]]:
     return _dict_list_to_list_dict(sample)
 
 
+def rearrange_samples(batch: List[Dict[str, Any]]) -> Dict[str, Any]:
+    """Define collate function to combine patches per tile and batch size.
+
+    Args:
+        batch: sample batch from dataloader containing image and mask
+
+    Returns:
+        sample batch where the batch dimension is
+        'batch_size' * 'num_patches_per_tile'
```

Comment:
This function does not "define a collate function", it _is_ a collate function. Also, rearrange is a bit ambiguous. Maybe something like:
```suggestion
def flatten_samples(samples: Iterable[Dict[str, Any]]) -> Dict[str, Any]:
    """Flatten the first two dimensions of a list of samples.
    
    Useful for ...
    
    Args:
        samples: list of samples
        
    Returns:
        a single sample
```

Reply:
Sounds good. Does that mean you also want to do `torch.flatten()` instead of the einops operation?

Reply:
Yes, for the reasons mentioned above. Explicit is good, but making assumptions about the names of keys and dimensions of keys is not.